### DISCUSSION TITLE
Generating empty pdf

### DISCUSSION THREAD

**Discussion by roobiniano** [dompdf/dompdf#2806] [Q&A]
Hello everyone, I have a problem with the library. Dompdf is generating empty PDF files, no matter what content I pass to the loadHtml method. The library was installed with composer, and I don't have any warnings or fatal errors in the logs.

Can you help me please ?

I will be very grateful

**My code:**

```
try {
    $dompdf = new Dompdf();
			
    $dompdf->setPaper('A4');

    $dompdf->loadHtml('ALOHA'); // $dompdf->loadHtml('<h1>ALOHA</h1>');
			
    // Render the HTML as PDF
    $dompdf->render();

     // Output the generated PDF to Browser
     $dompdf->stream();

 } catch (\Exception $e) {
     return $this->responseError($e->getMessage());
}
```

**Comment by bsweeney** [comment]
Your sample code looks fine. Do you have a sample of the PDF?

**Reply by roobiniano** [reply]
Hi @bsweeney , thanks for the answer, do you mean the binary code of the pdf file ?

**Reply by roobiniano** [reply]
%PDF-1.7
1 0 obj
<< /Type /Catalog
/Outlines 2 0 R
/Pages 3 0 R >>
endobj
2 0 obj
<< /Type /Outlines /Count 0 >>
endobj
3 0 obj
<< /Type /Pages
/Kids [6 0 R
]
/Count 1
/Resources <<
/ProcSet 4 0 R
/Font << 
/F1 8 0 R
>>
>>
/MediaBox [0.000 0.000 595.280 841.890]
 >>
endobj
4 0 obj
[/PDF /Text ]
endobj
5 0 obj
<<
/Producer (þÿdompdf 1.2.0 + CPDF)
/CreationDate (D:20220316111042-03'00')
/ModDate (D:20220316111042-03'00')
>>
endobj
6 0 obj
<< /Type /Page
/MediaBox [0.000 0.000 595.280 841.890]
/Parent 3 0 R
/Contents 7 0 R
>>
endobj
7 0 obj
<< /Filter /FlateDecode
/Length 68 >>
stream
xã2Ð300P@&Ò¹BMôÍÌ-LôLÌ,BRôÝôBÒ¢5}ü=5cB¼\Cøì
endstream
endobj
8 0 obj
<< /Type /Font
/Subtype /Type1
/Name /F1
/BaseFont /Times-Roman
/Encoding /WinAnsiEncoding
>>
endobj
xref
0 9
0000000000 65535 f 
0000000009 00000 n 
0000000074 00000 n 
0000000120 00000 n 
0000000274 00000 n 
0000000303 00000 n 
0000000452 00000 n 
0000000555 00000 n 
0000000694 00000 n 
trailer
<<
/Size 9
/Root 1 0 R
/Info 5 0 R
/ID[<fb6f048d2a5b6163c90145f5201c548e><fb6f048d2a5b6163c90145f5201c548e>]
>>
startxref
803
%%EOF

**Reply by roobiniano** [reply]
[dompdf_test.pdf](https://github.com/dompdf/dompdf/files/8262884/dompdf_test.pdf)

@bsweeney 
here is the file

**Reply by bsweeney** [reply]
That certainly is an empty PDF. The PDF is well-formed so I don't see anything indicating an issue during rendering. What version of Dompdf?

**Reply by roobiniano** [reply]
"dompdf/dompdf": "^1.2" 
installed via composer, the app runs on an IIS server.

<img width="616" alt="Captura de Pantalla 2022-03-16 a la(s) 23 24 59" src="https://user-images.githubusercontent.com/5823589/158724118-476340c6-f16c-43c7-8c9e-acbb8989b469.png">

**Reply by bsweeney** [reply]
Based on your try/catch and the directory structure you're running in the context of another framework. Could something there be interfering with PDF generation?

**Reply by roobiniano** [reply]
I am doing **use \Dompdf\Dompdf;**  at the beginning of the class

it's an MVC framework, very simple, I have no idea what could be interfering, I'm confused not to have any warning, error or notice in the logs

**Reply by bsweeney** [reply]
The fact that there is nothing logged just means the system is operating normally. Which is why I'm wondering if something is intercepting the request. One thing you can try is to call `exit` after `$dompdf->stream()`. That will halt any further processing, which should be unnecessary since you've just set the HTTP headers and sent relevant content.

Also, are you sure the Dompdf logic is actually running? Maybe add a logging statement within there to confirm.

**Reply by roobiniano** [reply]
yes I already tried with exit after the stream
this is a var_dump from dompdf->getOptions

`object(Dompdf\Options)#15 (27) { ["rootDir":"Dompdf\Options":private]=> string(53) "C:\inetpub\wwwroot\desarollo\mde\vendor\dompdf\dompdf" ["tempDir":"Dompdf\Options":private]=> string(15) "C:\Windows\TEMP" ["fontDir":"Dompdf\Options":private]=> string(63) "C:\inetpub\wwwroot\desarollo\mde\vendor\dompdf\dompdf/lib/fonts" ["fontCache":"Dompdf\Options":private]=> string(63) "C:\inetpub\wwwroot\desarollo\mde\vendor\dompdf\dompdf/lib/fonts" ["chroot":"Dompdf\Options":private]=> array(1) { [0]=> string(53) "C:\inetpub\wwwroot\desarollo\mde\vendor\dompdf\dompdf" } ["logOutputFile":"Dompdf\Options":private]=> string(23) "C:\Windows\TEMP/log.htm" ["defaultMediaType":"Dompdf\Options":private]=> string(6) "screen" ["defaultPaperSize":"Dompdf\Options":private]=> string(6) "letter" ["defaultPaperOrientation":"Dompdf\Options":private]=> string(8) "portrait" ["defaultFont":"Dompdf\Options":private]=> string(5) "serif" ["dpi":"Dompdf\Options":private]=> int(96) ["fontHeightRatio":"Dompdf\Options":private]=> float(1.1) ["isPhpEnabled":"Dompdf\Options":private]=> bool(false) ["isRemoteEnabled":"Dompdf\Options":private]=> bool(false) ["isJavascriptEnabled":"Dompdf\Options":private]=> bool(true) ["isHtml5ParserEnabled":"Dompdf\Options":private]=> bool(false) ["isFontSubsettingEnabled":"Dompdf\Options":private]=> bool(true) ["debugPng":"Dompdf\Options":private]=> bool(false) ["debugKeepTemp":"Dompdf\Options":private]=> bool(false) ["debugCss":"Dompdf\Options":private]=> bool(false) ["debugLayout":"Dompdf\Options":private]=> bool(false) ["debugLayoutLines":"Dompdf\Options":private]=> bool(true) ["debugLayoutBlocks":"Dompdf\Options":private]=> bool(true) ["debugLayoutInline":"Dompdf\Options":private]=> bool(true) ["debugLayoutPaddingBox":"Dompdf\Options":private]=> bool(true) ["pdfBackend":"Dompdf\Options":private]=> string(4) "CPDF" ["pdflibLicense":"Dompdf\Options":private]=> string(0) "" }`

**Reply by roobiniano** [reply]
the log.htm content

`<span style='color: #000' title='Frames'>     4</span><span style='color: #009' title='Memory'>   4096.00 KB</span><span style='color: #900' title='Time'>     35.28 ms</span><span  title='Quirksmode'>  <span style='color: #d00'> ON</span></span><br />`

**Reply by roobiniano** [reply]
I have the feeling that it is a problem with the fonts, because the pdf creates pages according to the length of the pdf content, the content is reaching the body of the pdf, but it is not printing it.

**Reply by bsweeney** [reply]
Normally a font problem would manifest differently, though it's certainly possible something like that is going on. If you add a border around your text does it render in the PDF?

```
$dompdf->loadHtml('<div style="border: 5px solid black;">ALOHA</div>');
```

**Reply by roobiniano** [reply]
no, it's not rendering borders either

**Reply by indexZ3ro** [reply]
I ran into this issue also, was there any solution?

**Reply by bsweeney** [reply]
I didn't see anything that stood out to me. Might take some user-side debugging to find the issue. The wiki has some info on how to [enable additional debug logging](https://github.com/dompdf/dompdf/wiki/Debugging-Dompdf).

You could even go one step further and use [callbacks](https://github.com/dompdf/dompdf/wiki/Usage#callbacks) to check that the text is actually present in the frames during render.
